https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/1024

From 5e3bf3d651e08cb7464f6c7270a9637239fc13bb Mon Sep 17 00:00:00 2001
From: John Zulauf <jzulauf@lunarg.com>
Date: Sat, 29 Jun 2019 07:55:57 -0600
Subject: [PATCH] layers: Use correct apiversion for properties check

Change layers property check from physical device apiVersion to the
effective apiVersion (the lesser of the device and CreateInstance
apiVersion values) to avoid invalid downchain call.

Change-Id: I282950f0f8927df280c97098e775a6940db809f1
---
 layers/core_validation.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git layers/core_validation.cpp layers/core_validation.cpp
index 03149c08d..75e473af1 100644
--- layers/core_validation.cpp
+++ layers/core_validation.cpp
@@ -2412,7 +2412,7 @@ void CoreChecks::PostCallRecordCreateDevice(VkPhysicalDevice gpu, const VkDevice
         instance_dispatch_table.GetPhysicalDeviceCooperativeMatrixPropertiesNV(gpu, &numCooperativeMatrixProperties,
                                                                                core_checks->cooperative_matrix_properties.data());
     }
-    if (core_checks->phys_dev_props.apiVersion >= VK_API_VERSION_1_1) {
+    if (core_checks->api_version >= VK_API_VERSION_1_1) {
         // Get the needed subgroup limits
         auto subgroup_prop = lvl_init_struct<VkPhysicalDeviceSubgroupProperties>();
         auto prop2 = lvl_init_struct<VkPhysicalDeviceProperties2KHR>(&subgroup_prop);
